src/models/cost_function_timeseries.jl

```diff
@@ -0,0 +1,108 @@
+function get_variable_cost(
+    ts::IS.TimeSeriesData,
+    component::Component,
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+)
+    if start_time === nothing
+        start_time = IS.get_initial_timestamp(ts)
+    end
+    data = IS.get_time_series_array(component, ts, start_time, len = len)
+    time_stamps = TimeSeries.timestamp(data)
+    return TimeSeries.TimeArray(time_stamps, map(VariableCost, TimeSeries.values(data)))
+end
+
+function get_variable_cost(
+    device::StaticInjection,
+    cost::OperationalCost;
+    start_time::Union{Nothing, Dates.DateTime} = nothing,
+    len::Union{Nothing, Int} = nothing,
+)
+    time_series_key = get_variable(cost)
+    if isnothing(time_series_key)
+        error("Cost component has a `nothing` stored in field `variable`, Please use `set_variable_cost!` to add variable cost forecast.")
+    end
+    raw_data = IS.get_time_series(
+        IS.time_series_metadata_to_data(time_series_key.time_series_type),
```

Comment:
The innards of TimeSeriesKey are getting spread out quite a bit.  Here are some possible improvements:
- Create a TimeSeriesKey constructor that accepts a TimeSeriesData.
- Create a get_time_series method that takes a TimeSeriesKey instead of type/name.

What do you think?  @sourabhdalvi @jd-lara

Reply:
I think it's better to have a get time series method that takes a time series key. The only con to that is that it might be exposed in the repl

Reply:
I wouldn't be opposed to `IS.get_time_series_internal`.

Reply:
I am fine with that and opens the question for later if should add `_internal` for private API.

Reply:
It could also be `get_time_series_by_key`.  In Julia that feels redundant because TimeSeriesKey is a function argument.  But that is a common pattern I've seen elsewhere.

Reply:
I think `IS.get_time_series_by_key` is the best approach. Even with the repetition, it is better.